datalad/core/distributed/clone.py

```diff
@@ -385,6 +386,8 @@ def __call__(
                 # from git default behavior WRT a submodule's name vs
                 # its path when we made this a new subdataset.
                 subds_name = path.relative_to(ds.pathobj)
+                # ensure posix url, relative Windows paths would be butchered
+                source = Path(source).as_posix() if not source.startswith('ria') else source
```

Comment:
I don't even think `ria` is an exclusion criterion here. Also: You are assuming `source` to always be a path (what is `Path('https://somewhere.org/blob')` supposed to do?)

I think the issue is with `source` already. If this was the spot to fix it, we would have cloned from something with `\` already. Which may work, but is not the intention really. So, in my view, source needs to be fixed before cloning if it happens to be a path. This may already happen somewhere in `clone_dataset`. There should be something that determines the `giturl` which is what is actually used and it's part of some dict (`something_props`?). I guess, this is what is supposed to be used here and possibly fixed for windows paths before.

Reply:
> Also: You are assuming source to always be a path (what is Path('https://somewhere.org/blob') supposed to do?)

I believe pathlib actually works well for non-paths - at least the cases I could think of. E.g., consider
```py
>>>from pathlib import Path
>>>Path('https://somewhere.org/blob').as_posix()
'https:/somewhere.org/blob'
>>>Path('git@github.com:datalad/datalad.git').as_posix()
'git@github.com:datalad/datalad.git'
>>>Path('adina@juseless.inm7.de:/home/scratch/weee').as_posix()
 'adina@juseless.inm7.de:/home/scratch/weee'
```

What else is a potential clone source beyond those URLs, local paths, and ria URLs?

Reply:
Re `ria`: The check with `startswith` is fine, the question is: What to do then? I think, the same should apply, since everything after the ria prefix is just a URL:

```
if source.startswith('ria'):
    source = 'ria+' + Path(source[4:]).as_posix()
else:
    source = Path(source).as_posix()
```

Reply:
> Did you try that on Windows? May be it works, too, in that `Path` would become a `PurePosixPath` or something, but I'm not sure.

This is on Windows 11 with Python 3.10
```
In [1]: from pathlib import Path 
In [2]: Path('https://somewhere.org/blob').as_posix()  
Out[2]: 'https:/somewhere.org/blob'                                                                                                                                                                                                            
In [3]: Path('git@github.com:datalad/datalad.git').as_posix()                                                           
Out[3]: 'git@github.com:datalad/datalad.git'                                                                                                                                                                                                    
In [4]: Path('adina@juseless.inm7.de:/home/scratch/weee').as_posix()                                                    
Out[4]: 'adina@juseless.inm7.de:/home/scratch/weee'                                                                                                                              
```

Reply:
C:\Users\mih\TMP\someds>datalad create-sibling-ria -s new --new-store-ok ria+file:/Users/mih/TMP/somestore
[INFO] create siblings 'new' and 'new-storage' ...
[INFO] Fetching updates for Dataset("C:\Users\mih\TMP\someds")
update(ok): . (dataset)
update(ok): . (dataset)
[INFO] Configure additional publication dependency on "new-storage"
configure-sibling(ok): . (sibling)
create-sibling-ria(ok): C:\Users\mih\TMP\someds (dataset)
action summary:
  configure-sibling (ok: 1)
  create-sibling-ria (ok: 1)
  update (ok: 1)
0.00 [00:04, ?/s]
C:\Users\mih\TMP\someds>git remote -v
new     \Users\mih\TMP\somestore\7b9\5bb8c-750f-4e2e-8e91-a083af8b2b13 (fetch)
new     \Users\mih\TMP\somestore\7b9\5bb8c-750f-4e2e-8e91-a083af8b2b13 (push)
new-storage
store   \Users\mih\TMP\RIA\7b9\5bb8c-750f-4e2e-8e91-a083af8b2b13 (fetch)
store   \Users\mih\TMP\RIA\7b9\5bb8c-750f-4e2e-8e91-a083af8b2b13 (push)
store-storage
C:\Users\mih\TMP\someds>datalad create-sibling-ria -s new2 --new-store-ok ria+file:\Users\mih\TMP\someotherstore
[INFO] create siblings 'new2' and 'new2-storage' ...
create-sibling-ria(error): C:\Users\mih\TMP\someds (dataset) [initremote failed.
stdout: initremote new2-storage
failed

stderr: git-annex: Non-absolute object tree base path configuration: \Users\mih\TMP\someotherstore
initremote: 1 failed
]
0.00 [00:03, ?/s]

Reply:
FTR: Using `file:` URLs on windows is a story on its own. That's unrelated to what's going on here. If you want to try `create-sibling-ria` with such a URL, it should be generated by `datalad.support.network.get_local_file_url(.., compatibility='git')` and then prefixed with `ria+`. That's not nice (and may still be buggy), but I don't see exactly how `create-sibling-ria` is supposed to fix a given URL that is incorrect.